decidim-elections/app/controllers/decidim/elections/elections_controller.rb

```diff
@@ -55,10 +55,18 @@ def search_klass
       def default_filter_params
         {
           search_text: "",
-          state: %w(active)
+          state: default_filter_state_params
         }
       end
 
+      def default_filter_state_params
+        filter_state_params = %w()
+        filter_state_params << "active" if elections.active
+        filter_state_params << "upcoming" if elections.upcoming && !elections.active
+        filter_state_params << "finished" if elections.finished && !elections.active && !elections.upcoming
+        filter_state_params
```

Comment:
I find this structure a little bit complex in this case, do you think that this `if` structure could be clearer?
```ruby
if elections.active
  %w(active)
elsif elections.upcoming
  %w(upcoming)
else # do we really need to check for finished elections here?
  %w()
end
```

Reply:
Yeah, maybe it doesn't make sense to filter for finished elections only. I updated it. 😊